File: runtimes/node/examples/bitcoin.js
#!/usr/bin/env node

var accessors = require('../accessors');

accessors.create_accessor('/webquery/Bitcoin', {}, function (err, accessor) {
	if (err) {
		console.log('Error loading accessor.');
		console.log(err);
		return;
	}

	accessor.init(function(err) {
		if (err) {
			console.log('Could not init accessor.');
			console.log(err);
			return;
		}

		accessor.read('Price', function (err, price) {
			if (err) {
				console.log('Some issue contacting the BTC server.');
				return;
			}

			console.log('Current Bitcoin Price: $' + price);
		});

		accessor.on('Transactions', function (err, transaction) {
			console.log('=== Bitcoin Transaction ===');
			for (var i=0; i<transaction.x.inputs.length; i++) {
				var input = transaction.x.inputs[i];
				var addr = input.prev_out.addr;
				var btc = (parseFloat(input.prev_out.value) / 100000000.0).toFixed(6);
				console.log('  FROM ' + addr + '  (' + btc + ' BTC)');
			}

			for (var i=0; i<transaction.x.out.length; i++) {
				var output = transaction.x.out[i];
				var addr = output.addr;
				var btc = (parseFloat(output.value) / 100000000.0).toFixed(6);
				console.log('  TO   ' + addr + '  (' + btc + ' BTC)');
			}

			console.log('');
		});

	});
});

File: runtimes/python/bitcoin.py
#!/usr/bin/env python3

import accessors
#import logging
#logging.getLogger('accessors').setLevel(logging.DEBUG)

def transaction_callback(transaction):
	print('=== Bitcoin Transaction ===');
	for trans in transaction['x']['inputs']:
		addr = trans['prev_out']['addr']
		btc = float(trans['prev_out']['value']) / 100000000.0
		print('  FROM {}  ({:.6f} BTC)'.format(addr, btc));

	for trans in transaction['x']['out']:
		addr = trans['addr']
		btc = float(trans['value']) / 100000000.0
		print('  TO   {}  ({:.6f} BTC)'.format(addr, btc));

	print('')


Bitcoin = accessors.get_accessor('/webquery/Bitcoin')

value = Bitcoin.Price
print("Bitcoin.Price = {}".format(value))

Bitcoin.observe('Transactions', transaction_callback)

accessors.observe_forever()

